src/mewbot/tools/dev_venv.py

```diff
@@ -0,0 +1,632 @@
+
+    # 1 - Build Venv
+    with BasicCommandDelimiter(tool_name="1 - Building Venv"):
+        _build_venv(venv_path)
+
+    # 2 - Validate venv
+    with BasicCommandDelimiter(tool_name="2 - Validate Venv"):
+        activation_path = _validate_venv(venv_path)
+
+    # 3 - Update venv pip
+    with BasicCommandDelimiter(tool_name="3 - Update venv pip"):
+        subprocess.run(
+            [
+                f"{activation_path}",
+                "&&",
+                "python",
+                "-m",
+                "pip",
+                "install",
+                "--upgrade",
+                "pip",
+            ],
+            shell=True,
+            check=True,
+        )
+
+    # 4 - Install all deps into the venv using the install_deps script
+    with BasicCommandDelimiter(tool_name="4 - Install all deps into the venv"):
+        _install_all_deps(mewbot_repo_path=MEWBOT_REPO_PATH, activation_path=activation_path)
+
+    # 5 - install the plugins
+    with BasicCommandDelimiter(tool_name="5 - install the plugins"):
+        _install_all_plugins(
+            mewbot_repo_path=MEWBOT_REPO_PATH,
+            activation_path=activation_path,
+            mewbot_plugins_dir=MEWBOT_PLUGINS_DIR,
+        )
+
+    # 6 - Wipe the placeholder installs for the mewbot submodules
+    #     These have installed from pypi and are no longer needed
+    with BasicCommandDelimiter(tool_name="6 - Wipe pypi placeholder installs"):
+        _wipe_pypi_installs(activation_path)
+
+    # 7 - Install the sub-modules in editable form
+    with BasicCommandDelimiter(tool_name="7 - Install the sub-modules in editable form"):
+        _install_local_submodules(activation_path)
+
+    # 8 - Move the egg-info folders somewhere sensible
+    with BasicCommandDelimiter(tool_name="8 - Rehome egg-info folders"):
+        _rehome_egg_info_folders(
+            src_path=SRC_PATH, plugins_path=MEWBOT_PLUGINS_DIR, venv_path=venv_path
+        )
+
+    # 9 - Run tests on the venv
+    with BasicCommandDelimiter(tool_name="9 - Run tests on the venv"):
+        _test_build_venv(activation_path)
+
+    # 10 - Tell the user we're done
+    print(f'\n\nvenv prepared - activate via "{activation_path}"')
+
+    # If we reach this point, everything should be okay
+    return True
+
+
+def _validate_input() -> str:
+    """
+    Confirm that the input from the command line is valid under the set criteria.
+
+    E.g. _hopefully_ not going to cause a really embarrassing security hole.
+    :return:
+    """
+    try:
+        target_dir: str = sys.argv[1]
+    except IndexError as exp:
+        raise NotImplementedError(
+            "You need to provide a path to a location to build the venv in"
+        ) from exp
+
+    target_dir = os.path.abspath(target_dir)
+    if not os.path.exists(target_dir):
+        raise NotImplementedError(f"{target_dir = } must exist.")
+
+    venv_name = sys.argv[2]
+    assert (
+        re.match(r"^[a-zA-Z0-9_]+$", venv_name) is not None
```

Comment:
hmmm.
https://pypi.org/project/pathvalidate/#filename-filepath-validator-for-argparse does a "better" check. My first instinct for testing of `venv-test` didn't match this, which I didn't expect. And then I realised I don't know what the best regexp would be.

Reply:
Ported in enough of their code to do a better validation. Credited and licensed under MIT.